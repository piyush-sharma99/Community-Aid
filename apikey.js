/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
