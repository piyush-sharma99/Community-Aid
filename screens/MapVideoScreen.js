/*
 *  ClassName: MapVideoScreen.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

//Imports
import React from "react";
import { StyleSheet, Text, View, ImageBackground } from "react-native";
import { WebView } from "react-native-webview";

//Main Component
const MapVideoScreen = (props) => {
  return (
    <ImageBackground source={require("../assets/BG.png")} style={styles.bImage}>
      <View style={styles.screen}>
        <View style={styles.subScreen}>
          <WebView
            style={styles.web}
            javaScriptEnabled={true}
            source={{ uri: "https://youtu.be/ZCSVUAULKWI" }}
          />
        </View>

        <View>
          <Text style={styles.text}>
            The map page is designed for volunteers who would like to help
            others in their region. It allows volunteers to view requests in
            their area, view their area, set a radius ring on the map to measure
            distance and add requests on demand. To view a tutorial on how to
            use the features view the video above.{" "}
          </Text>
        </View>
      </View>
    </ImageBackground>
  );
};

//Navigation options: changing header displayed on the page
MapVideoScreen.navigationOptions = {
  headerTitle: "Map Assistance ",
  headerStyle: {
    backgroundColor: "#2c8ffa",
  },
  headerTintColor: "white",
};

//CSS
const styles = StyleSheet.create({
  bImage: {
    width: "100%",
    height: "100%",
  },

  screen: {
    flex: 1,
  },
  subScreen: {
    height: "50%",
  },

  web: {
    marginRight: 5,
    marginLeft: 5,
    marginTop: 30,
    borderWidth: 2,
    borderColor: "#fb5b5a",
    elevation: 10,
  },

  text: {
    padding: 10,
    marginRight: 40,
    marginTop: 40,
    marginBottom: 30,
    marginLeft: 40,
    color: "white",
    fontSize: 20,
    textAlign: "center",
    backgroundColor: "#fb5b5a",
    borderWidth: 3,
    borderColor: "white",
    shadowOpacity: 2,
    elevation: 5,
    shadowOffset: {
      width: 3,
      height: 3,
    },
  },
});

export default MapVideoScreen;
