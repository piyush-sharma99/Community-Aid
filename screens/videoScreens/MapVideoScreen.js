import React from 'react';
import { StyleSheet, Text, View,} from 'react-native';
import { WebView } from 'react-native-webview';


const MapVideoScreen = props => {
    return(
        <View style={styles.screen}>
            <View style={styles.subScreen}>
              <WebView
    style={styles.web}
    javaScriptEnabled={true}
    source={{uri: 'https://youtu.be/ZCSVUAULKWI'}}
/>

</View>

<View>
    <Text style={styles.text}>The map page is designed for volunteers who would like to help others in their region. It allows volunteers to view requests in their area, view their area, set a radius ring on the map to measure distance and add requests on demand. To view a tutorial on how to use the features view the video above. </Text>
</View>
    
        </View>
    );
};

MapVideoScreen.navigationOptions = {
    headerTitle: 'Map Assistance ',
    headerStyle: {
    backgroundColor: '#2c8ffa'
    },
    headerTintColor:"white"
    };

    const styles = StyleSheet.create({
        screen: {
            flex: 1,
            backgroundColor: '#2c8ffa',
            
        },
        subScreen: {
            height:'50%',
            backgroundColor: '#2c8ffa',
                
        },
    

          web:{
            marginRight:5,
            marginLeft:5,
            marginTop:30,
            borderWidth:2,
            borderColor: '#fb5b5a',
            elevation: 10
            
            
            
          },

          text: {
            padding:10,
            marginRight:40,
            marginTop:40,
            marginBottom:30,
            marginLeft:40,
            color:"white",
            fontSize:20,
            textAlign: 'center',
            backgroundColor:'#fb5b5a',
            borderWidth: 3,
            borderColor: 'white',
            shadowOpacity: 2,
        elevation: 5,
        shadowOffset: {
          width: 3,
          height: 3
      }
            
        
        },
    });

export default MapVideoScreen;