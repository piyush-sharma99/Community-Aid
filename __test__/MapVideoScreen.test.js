import React from "react";
import MapVideoScreen from "../screens/MapVideoScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("MapVideo renders correctly", () => {
  const tree = renderer.create(<MapVideoScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
