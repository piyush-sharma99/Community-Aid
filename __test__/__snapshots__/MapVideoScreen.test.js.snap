// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapVideo renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <RNCWebView
          cacheEnabled={true}
          injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
          injectedJavaScriptForMainFrameOnly={true}
          javaScriptEnabled={true}
          messagingEnabled={false}
          onContentProcessDidTerminate={[Function]}
          onHttpError={[Function]}
          onLoadingError={[Function]}
          onLoadingFinish={[Function]}
          onLoadingProgress={[Function]}
          onLoadingStart={[Function]}
          onMessage={[Function]}
          onShouldStartLoadWithRequest={[Function]}
          source={
            Object {
              "uri": "https://youtu.be/ZCSVUAULKWI",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#ffffff",
              },
              Object {
                "borderColor": "#fb5b5a",
                "borderWidth": 2,
                "elevation": 10,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 30,
              },
            ]
          }
          useSharedProcessPool={true}
        />
      </View>
    </View>
    <View>
      <Text
        style={
          Object {
            "backgroundColor": "#fb5b5a",
            "borderColor": "white",
            "borderWidth": 3,
            "color": "white",
            "elevation": 5,
            "fontSize": 20,
            "marginBottom": 30,
            "marginLeft": 40,
            "marginRight": 40,
            "marginTop": 40,
            "padding": 10,
            "shadowOffset": Object {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 2,
            "textAlign": "center",
          }
        }
      >
        The map page is designed for volunteers who would like to help others in their region. It allows volunteers to view requests in their area, view their area, set a radius ring on the map to measure distance and add requests on demand. To view a tutorial on how to use the features view the video above.
         
      </Text>
    </View>
  </View>
</View>
`;
